source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/IValidationRule.cs

```diff
@@ -28,5 +28,10 @@ public interface IValidationRule
         /// Identifier of the current rule
         /// </summary>
         public ValidationRuleIdentifier ValidationRuleIdentifier { get; }
+
+        /// <summary>
+        /// Element in list that violated the rule identified by <see cref="ValidationRuleIdentifier"/>
+        /// </summary>
+        public string? ListElement { get; }
```

Comment:
How about this and remove all your null assignments in all derivatives?

```suggestion
        public string? ListElement { get; } = null;
```

Reply:
I gave it a chance but "[CS8053] Instance properties in interfaces cannot have initializers"

Reply:
How about introducing an abstract class? There seems to be shared (default) functionality in most of the validation rules.
(possibly just making the interface abstract)?